### PR TITLE
Support reparenting state from MultiChildRenderObjectElement

### LINKED ISSUE #2085: RenderObjectElement.updateChildren() doesn't support GlobalKeys
Right now I believe it does.

### PR DESCRIPTION
**Pull request by abarth**
Fixes #2085

### CODE REVIEW

**Comment by abarth** [comment]
@Hixie

File: packages/flutter/lib/src/widgets/framework.dart [hunk 1]
```diff
@@ -1605,8 +1605,8 @@ abstract class RenderObjectElement<T extends RenderObjectWidget> extends Buildab
     while ((oldChildrenTop <= oldChildrenBottom) && (newChildrenTop <= newChildrenBottom)) {
```
Comment (Hixie) [inline packages/flutter/lib/src/widgets/framework.dart]: This function has other customers; I guess they need to handle detachChild also in due course.

Comment (abarth) [inline packages/flutter/lib/src/widgets/framework.dart]: Yeah, the changes here shouldn't be harmful to them, but they don't all support detachChild yet.

**Comment by Hixie** [comment]
lgtm. I'm not sure I understood the test, I used up all my brainpower on the updateChildren change.

**Pull request merged**